tripleo/utils/git/utils.py

```diff
@@ -0,0 +1,47 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+from urllib.parse import urlparse
+
+from tripleo.utils.types import URL
+from tripleo.utils.urls import is_git
+
+
+def get_repository_fullname(url: URL) -> str:
+    """Gets the fullname of a repository from its URL.
+
+    URL formats supported are:
+        - HTTP/HTTPS
+        - SSH
+
+    Examples
+    --------
+    >>> cibyl = URL('https://github.com/rhos-infra/cibyl.git')
+    ... get_repository_fullname(cibyl)
+    'rhos-infra/cibyl'
+
+    :param url: The URL to get the data from.
+    :return: The fullname in the format: '{owner}/{name}'
+    :raises ValueError: If the URL does not point to a git repository.
+    """
+    if not is_git(url):
+        msg = f"URL must point to a git repository: '{url}'."
+        raise ValueError(msg)
+
+    path = urlparse(url).path
+    path = path[1:]  # Remove leading '/'
+    path = path[:-4]  # Remove trailing '.git'
```

Comment:
The URL that I expect here is the one GitHub gives you in order to clone the repository, be it through HTTP or SSH. What I do not know now is if that URL can indeed not have the suffix. I thought it would.

Reply:
I'm not sure. The cloning url, and the one stored for the remotes, does have the '.git' extension, but the browsable ones don't. That's why I mentioned that it could depend on the source. Since the check should be simple, IMO it would be good to add it just in case.

Reply:
On any case, just before these lines, 'is_git' will raise an error if the URL has no suffix. It is consistent in that assumption at least.

This would mean that at this point Cibyl only accepts URLs ending in '.git. Being that consistent, I think it could be left for a future task to expand that.

Reply:
Oh I missed that check, then yeah, no issue with that, you can disregard my point completely, sorry!